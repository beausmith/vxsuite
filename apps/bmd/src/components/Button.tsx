import styled from 'styled-components'

const Button = styled('button')<JSX.IntrinsicElements['button']>`
  background: lightgrey;
  border-radius: 0.25rem;
  color: black;
  white-space: nowrap;
`
export default Button
