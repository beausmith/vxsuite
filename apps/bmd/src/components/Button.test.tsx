import { axe } from 'jest-axe'
import React from 'react'
import { render } from 'react-testing-library'

import Button from './Button'

it(`renders Button`, async () => {
  const { container } = render(<Button>foo</Button>)
  expect(await axe(container.innerHTML)).toHaveNoViolations()
  expect(container.firstChild).toMatchSnapshot()
})
