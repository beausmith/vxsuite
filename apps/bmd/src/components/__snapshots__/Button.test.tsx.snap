// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Button 1`] = `
.c0 {
  background: lightgrey;
  border-radius: 0.25rem;
  color: black;
  white-space: nowrap;
}

<button
  class="c0"
>
  foo
</button>
`;
